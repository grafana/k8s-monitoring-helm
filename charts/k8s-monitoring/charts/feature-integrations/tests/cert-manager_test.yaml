# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test cert-manager integration
templates:
  - configmap.yaml
tests:
  - it: should create the cert-manager config
    set:
      deployAsConfigMap: true
      cert-manager:
        instances:
          - name: cert-manager
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: data["metrics.alloy"]
          value: |-
            declare "cert_manager_integration" {
              argument "metrics_destinations" {
                comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
              }
            
              import.git "cert_manager" {
                repository = "https://github.com/grafana/alloy-modules.git"
                revision = "main"
                path = "modules/kubernetes/cert-manager/metrics.alloy"
                pull_frequency = "15m"
              }
            
              cert_manager.kubernetes "cert_manager" {
                label_selectors = ["app.kubernetes.io/name=cert-manager"]
                port_name = "http-metrics"
              }
            
              cert_manager.scrape "cert_manager" {
                targets = cert_manager.kubernetes.cert_manager.output
                clustering = true
                job_label = "integrations/cert-manager"
                scrape_interval = "60s"
                max_cache_size = 100000
                forward_to = argument.metrics_destinations.value
              }
            }
  - it: can be restricted to a namespace
    set:
      deployAsConfigMap: true
      cert-manager:
        instances:
          - name: my-cert-manager
            namespaces:
              - kube-system
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: data["metrics.alloy"]
          value: |-
            declare "cert_manager_integration" {
              argument "metrics_destinations" {
                comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
              }
            
              import.git "cert_manager" {
                repository = "https://github.com/grafana/alloy-modules.git"
                revision = "main"
                path = "modules/kubernetes/cert-manager/metrics.alloy"
                pull_frequency = "15m"
              }
            
              cert_manager.kubernetes "my_cert_manager" {
                namespaces = ["kube-system"]
                label_selectors = ["app.kubernetes.io/name=my-cert-manager"]
                port_name = "http-metrics"
              }
            
              cert_manager.scrape "my_cert_manager" {
                targets = cert_manager.kubernetes.my_cert_manager.output
                clustering = true
                job_label = "integrations/cert-manager"
                scrape_interval = "60s"
                max_cache_size = 100000
                forward_to = argument.metrics_destinations.value
              }
            }
