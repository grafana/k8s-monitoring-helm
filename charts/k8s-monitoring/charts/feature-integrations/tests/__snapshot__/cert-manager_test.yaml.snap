can be restricted to a namespace:
  1: |
    |-
      declare "cert_manager_integration" {
        argument "metrics_destinations" {
          comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
        }

        discovery.kubernetes "my_cert_manager" {
          role = "pod"

          selectors {
            role = "pod"
            label = "app.kubernetes.io/name=my-cert-manager"
          }
          namespaces {
            names = ["kube-system"]
          }
        }

        discovery.relabel "my_cert_manager" {
          targets = discovery.kubernetes.my_cert_manager.targets

          // keep only the specified metrics port name, and pods that are Running and ready
          rule {
            source_labels = [
              "__meta_kubernetes_pod_container_port_name",
              "__meta_kubernetes_pod_phase",
              "__meta_kubernetes_pod_ready",
            ]
            separator = "@"
            regex = "http-metrics@Running@true"
            action = "keep"
          }

          // drop any init containers
          rule {
            source_labels = ["__meta_kubernetes_pod_container_init"]
            regex = "true"
            action = "drop"
          }

          // set the namespace label
          rule {
            source_labels = ["__meta_kubernetes_namespace"]
            target_label  = "namespace"
          }

          // set the pod label
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            target_label  = "pod"
          }

          // set the container label
          rule {
            source_labels = ["__meta_kubernetes_pod_container_name"]
            target_label  = "container"
          }

          // set a workload label
          rule {
            source_labels = [
              "__meta_kubernetes_pod_controller_kind",
              "__meta_kubernetes_pod_controller_name",
            ]
            separator = "/"
            target_label  = "workload"
          }
          // remove the hash from the ReplicaSet
          rule {
            source_labels = ["workload"]
            regex = "(ReplicaSet/.+)-.+"
            target_label  = "workload"
          }

          // set the app name if specified as metadata labels "app:" or "app.kubernetes.io/name:" or "k8s-app:"
          rule {
            action = "replace"
            source_labels = [
              "__meta_kubernetes_pod_label_app_kubernetes_io_name",
              "__meta_kubernetes_pod_label_k8s_app",
              "__meta_kubernetes_pod_label_app",
            ]
            separator = ";"
            regex = "^(?:;*)?([^;]+).*$"
            replacement = "$1"
            target_label = "app"
          }

          // set the component if specified as metadata labels "component:" or "app.kubernetes.io/component:" or "k8s-component:"
          rule {
            action = "replace"
            source_labels = [
              "__meta_kubernetes_pod_label_app_kubernetes_io_component",
              "__meta_kubernetes_pod_label_k8s_component",
              "__meta_kubernetes_pod_label_component",
            ]
            regex = "^(?:;*)?([^;]+).*$"
            replacement = "$1"
            target_label = "component"
          }

          // set a source label
          rule {
            action = "replace"
            replacement = "kubernetes"
            target_label = "source"
          }
        }

        prometheus.scrape "my_cert_manager" {
          targets = discovery.relabel.my_cert_manager.output
          job_name = "integrations/cert-manager"
          scrape_interval = "60s"
          clustering {
            enabled = true
          }
          forward_to = argument.metrics_destinations.value
        }
      }
should create the cert-manager config:
  1: |
    |-
      declare "cert_manager_integration" {
        argument "metrics_destinations" {
          comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
        }

        discovery.kubernetes "cert_manager" {
          role = "pod"

          selectors {
            role = "pod"
            label = "app.kubernetes.io/name=cert-manager"
          }
        }

        discovery.relabel "cert_manager" {
          targets = discovery.kubernetes.cert_manager.targets

          // keep only the specified metrics port name, and pods that are Running and ready
          rule {
            source_labels = [
              "__meta_kubernetes_pod_container_port_name",
              "__meta_kubernetes_pod_phase",
              "__meta_kubernetes_pod_ready",
            ]
            separator = "@"
            regex = "http-metrics@Running@true"
            action = "keep"
          }

          // drop any init containers
          rule {
            source_labels = ["__meta_kubernetes_pod_container_init"]
            regex = "true"
            action = "drop"
          }

          // set the namespace label
          rule {
            source_labels = ["__meta_kubernetes_namespace"]
            target_label  = "namespace"
          }

          // set the pod label
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            target_label  = "pod"
          }

          // set the container label
          rule {
            source_labels = ["__meta_kubernetes_pod_container_name"]
            target_label  = "container"
          }

          // set a workload label
          rule {
            source_labels = [
              "__meta_kubernetes_pod_controller_kind",
              "__meta_kubernetes_pod_controller_name",
            ]
            separator = "/"
            target_label  = "workload"
          }
          // remove the hash from the ReplicaSet
          rule {
            source_labels = ["workload"]
            regex = "(ReplicaSet/.+)-.+"
            target_label  = "workload"
          }

          // set the app name if specified as metadata labels "app:" or "app.kubernetes.io/name:" or "k8s-app:"
          rule {
            action = "replace"
            source_labels = [
              "__meta_kubernetes_pod_label_app_kubernetes_io_name",
              "__meta_kubernetes_pod_label_k8s_app",
              "__meta_kubernetes_pod_label_app",
            ]
            separator = ";"
            regex = "^(?:;*)?([^;]+).*$"
            replacement = "$1"
            target_label = "app"
          }

          // set the component if specified as metadata labels "component:" or "app.kubernetes.io/component:" or "k8s-component:"
          rule {
            action = "replace"
            source_labels = [
              "__meta_kubernetes_pod_label_app_kubernetes_io_component",
              "__meta_kubernetes_pod_label_k8s_component",
              "__meta_kubernetes_pod_label_component",
            ]
            regex = "^(?:;*)?([^;]+).*$"
            replacement = "$1"
            target_label = "component"
          }

          // set a source label
          rule {
            action = "replace"
            replacement = "kubernetes"
            target_label = "source"
          }
        }

        prometheus.scrape "cert_manager" {
          targets = discovery.relabel.cert_manager.output
          job_name = "integrations/cert-manager"
          scrape_interval = "60s"
          clustering {
            enabled = true
          }
          forward_to = argument.metrics_destinations.value
        }
      }
